src/IssueViz.Security.UnitTests/Taint/TaintList/TaintIssuesControlViewModelTests.cs

```diff
@@ -439,14 +466,155 @@ public void HasServerIssues_IssuesChanged_RaisesPropertyChanged()
+
+            eventHandler.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public void SelectionChanged_SelectedIssueIsNotInList_RaisesPropertyChanged(bool isSelectedNull)
+        {
+            var selectionService = new Mock<IIssueSelectionService>();
+
+            var testSubject = CreateTestSubject(selectionService: selectionService.Object);
+
+            var eventHandler = new Mock<PropertyChangedEventHandler>();
+            testSubject.PropertyChanged += eventHandler.Object;
+
+            eventHandler.VerifyNoOtherCalls();
+
+            var selectedIssue = isSelectedNull ? null : Mock.Of<IAnalysisIssueVisualization>();
+
+            RaiseSelectionChangedEvent(selectionService, selectedIssue);
+
+            eventHandler.Verify(x => x(testSubject,
+                It.Is((PropertyChangedEventArgs args) =>
+                    args.PropertyName == nameof(testSubject.SelectedIssue))), Times.Once);
+
+            eventHandler.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        [Microsoft.VisualStudio.TestTools.UnitTesting.Description("Verify that there is no callback, selection service -> property -> selection service")]
+        public void SelectionChanged_IssueSelected_SelectionServiceNotCalledAgain()
+        {
+            var selectedIssue = Mock.Of<IAnalysisIssueVisualization>();
+            var storeIssues = new [] { selectedIssue };
+            var selectionService = new Mock<IIssueSelectionService>();
+
+            CreateTestSubject(storeIssues, selectionService: selectionService.Object);
+
+            RaiseSelectionChangedEvent(selectionService, selectedIssue);
+
+            selectionService.VerifySet(x => x.SelectedIssue = It.IsAny<IAnalysisIssueVisualization>(), Times.Never);
+        }
+
+        [TestMethod]
+        public void SetSelectedIssue_IssueSet()
+        {
+            var testSubject = CreateTestSubject();
+
+            var selection = new TaintIssueViewModel(Mock.Of<IAnalysisIssueVisualization>());
+            testSubject.SelectedIssue = selection;
+
+            testSubject.SelectedIssue.Should().Be(selection);
+        }
+
+        [TestMethod]
+        public void SetSelectedIssue_SelectionServiceIsCalled()
+        {
+            var selectionService = new Mock<IIssueSelectionService>();
+            var testSubject = CreateTestSubject(selectionService: selectionService.Object);
+
+            selectionService.Reset();
+
+            var selection = new TaintIssueViewModel(Mock.Of<IAnalysisIssueVisualization>());
+            testSubject.SelectedIssue = selection;
+
+            selectionService.VerifySet(x => x.SelectedIssue = selection.TaintIssueViz, Times.Once);
+            selectionService.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void SetSelectedIssue_ValueIsTheSame_SelectionServiceNotCalled()
```

Comment:
Is there a test that the selection service is called when the service changes from (something) -> null?